starport/interface/cli/starport/cmd/network_chain_join.go

```diff
@@ -106,85 +110,40 @@ func networkChainJoinHandler(cmd *cobra.Command, args []string) error {
 	}
 
 	// ask to create an account on target blockchain.
+	shouldCreateAccount := true
+
 	printSection(fmt.Sprintf("Account on the blockchain %s", chainID))
-	account, err = createChainAccount(cmd.Context(), blockchain, fmt.Sprintf("%s blockchain", chainID), accountName)
-	if err != nil {
-		return err
+
+	if gentxPath != "" {
+		askAccountProposal := promptui.Prompt{
+			Label:     "Would you like to propose an account to place in Genesis",
+			IsConfirm: true,
+		}
+		_, err := askAccountProposal.Run()
+		shouldCreateAccount = err == nil
 	}
 
-	// ask to create an account proposal,
-	printSection("Account proposal")
+	if shouldCreateAccount {
+		acc, err := createChainAccount(cmd.Context(), blockchain, fmt.Sprintf("%s blockchain", chainID), accountName)
```

Comment:
Of course, we still prompt the account coins since this value is not in the gentx

Reply:
This is what we're doing already, can you please re-check?

Account is being created if gentx is provided and user wants to. Is this flow wrong, what am I missing?

Reply:
I have a gentx with those value:
```
"delegator_address": "cosmos16kjdtah4z4wjv54yx7ph0dydnsx2zskfywdrna",
"validator_address": "cosmosvaloper16kjdtah4z4wjv54yx7ph0dydnsx2zskfp6eklw",
```
The account address for the `AddAccountProposal` should be directly `cosmos16kjdtah4z4wjv54yx7ph0dydnsx2zskfywdrna`

Currently if I run `starport network chain join mars --gentx ./gentx.json`
```
✔ Source code fetched
✔ Blockchain initialized
---------------------------------------------
Account on the blockchain mars
---------------------------------------------

Would you like to propose an account to place in Genesis: y
? Choose an account:  [Use arrows to move, type to filter]
> Create a new account
  Import an account from mnemonic
```

Maybe there was a misunderstanding on the notion of "creating an account". When doing the manual join we never create a new account on the host because the fact that the user already has a gentx ready means they already created their account. We just propose to the user if they want to create an `AddAccountProposal` along with the `AddValidatorProposal`

Reply:
Is it OK to always assume that validator's address will be same as account's address in the `join` cmd? I guess for now yes since `join` cmd is all about joining as a validator? And in the future we can have a `add-account` cmd as discussed to independently place arbitrary accounts in the genesis.

In that case, yes, seems like there is no reason to prompt the account dialog when `-gentx` flag is used unless user wants to import an existing account's mnemonic for that address.

Any final thoughts?

Reply:
The current `AddValidator` proposal always comes with a self delegation. It means the validator of the gentx will match an account in the genesis account array of the genesis. (we may discuss in the future a proposal where the validator is not created with a self delegation but it's likely to be another proposal type)
So yes, if a genesis account is added, this is always an account with the address of the validator.

The prompt should ask if the user wants to add the genesis account in an `AddAccount` proposal along with the `AddValidator` proposal. Since we can provide a custom genesis with preconfigured genesis accounts, the genesis account of the validator might be already present, therefore the user could answer no to the question, to only add the gentx.

Another way to go could be to verify if the genesis account is already present in the genesis and in this case we automatically add only an `AddValidator` proposal otherwise we add `AddAccount` and `AddValidator` but it might be too much automation, a user using the manual launch workflow should know what they're doing.

In any case, with the manual launch, there is never any side effect to the host machine keyring, no key is created or imported.

Reply:
I would go with the second option and auto add the account proposal and inform the user about it. (but maybe not subject of this PR)

> In any case, with the manual launch, there is never any side effect to the host machine keyring, no key is created or imported.

Why is this the case? As a user, I might have a wallet already and would like to use it for `join` while adding my account in the host machine's keyring for the chain. Isn't this a valid use case?

Reply:
changed to proposal confirmation in  2b02d13.

Reply:
> I would go with the second option and auto add the account proposal and inform the user about it. (but maybe not subject of this PR)

Yes it should definitely done in another PR but I'm still a bit skeptical about adding too much automation in the manual launch process. The fact that a user doesn't want to add an account but that is actually not present in the genesis can mean that something is wrong with the genesis and the user should be warned about this

> Why is this the case? As a user, I might have a wallet already and would like to use it for join while adding my account in the host machine's keyring for the chain. Isn't this a valid use case?

Having the gentx mean that the user signed it, therefore the user a;ready has the account somewhere. Since manual launch is for advanced usage, I think in lot of case the user would protect their account with a solution like a Ledger, etc... the user would not want to import the account the keyring that can be less secure